controllers/articles.js

```diff
@@ -239,6 +239,150 @@ const articles = {
       }
     }
   },
+
+  /*
+  * Bookmarking and article so that one can come back
+  * and read it later.
+  */
+  createBookmark: async (req, res) => {
+    try {
+      const { slug } = req.params;
+      const { id: userId } = req.user;
+      if (!userId) {
+        res.status(401).send({
+          status: res.statusCode,
+          errorMessage: 'Please first login to bookmark this article',
+        });
+      }
+      const findArticle = await article.findOne({ where: { slug } });
+      const findUser = await User.findOne({ where: { id: findArticle.author } });
+      const checkExist = await bookmark.findOne({ where: { articleId: findArticle.id, userId } });
+      if (checkExist) {
+        return res.status(400).send({
+          status: res.statusCode,
+          errorMessage: 'You have already bookmarked this article',
+        });
+      }
+
+      const createBookmark = await bookmark.create({
+        userId, articleId: findArticle.id
+      });
+      if (createBookmark.articleId) {
+        return res.status(200).send({
+          status: res.statusCode,
+          message: `Article from ${findUser.username} has been bookmarked`,
+          data: {
+            title: findArticle.title,
+            body: findArticle.body,
+            description: findArticle.description,
+          }
+        });
+      }
+      return res.status(404).send({
+        status: res.statusCode,
+        errorMessage: 'The article your trying to bookmark does not exit',
+      });
+    } catch (error) {
+      if (error.message) {
+        return res.status(500).send({
+          status: res.statusCode,
+          errorMessage: 'Something went wrong',
+        });
+      }
+    }
+  },
+
+  getAllBookmarks: async (req, res) => {
+    const { id: userId } = req.user;
+    try {
+      const bookmarkedArticles = [];
+      const findBookmarks = await bookmark.findAll({ where: { userId } });
+
+      // eslint-disable-next-line no-restricted-syntax
+      for (const item of findBookmarks) {
+        // eslint-disable-next-line no-await-in-loop
+        const getBookmark = await article.findOne({ where: { id: item.articleId }, attributes: ['title', 'body', 'description', 'slug', 'createdAt', 'updatedAt', 'categories', 'tagList', 'author'] });
+        // eslint-disable-next-line no-await-in-loop
+        const author = await User.findOne({ where: { id: getBookmark.author }, attributes: ['username', 'image'] });
+        getBookmark.author = author;
+        bookmarkedArticles.push(getBookmark);
+      }
+
+      if (findBookmarks) {
+        return res.status(200).send({
+          status: res.statusCode,
+          data: bookmarkedArticles
```

Comment:
@aaronsekisambu while testing this, I have noticed that in return object instead of returning the object as the article itself it is called Bookmark. I think you can change it.